src/main/java/eu/dissco/backend/service/AnnotationService.java

```diff
@@ -134,19 +140,20 @@ public AnnotationResponse getAnnotationByVersion(String id, int version)
   }
 
   private AnnotationResponse mapToAnnotation(JsonNode result) {
+    var annotation = result.get("annotation");
     return new AnnotationResponse(
         result.get("id").asText(),
         result.get("version").asInt(),
-        result.get("type").asText(),
-        result.get("motivation").asText(),
-        result.get("target_body"),
-        result.get("body"),
-        result.get("preference_score").asInt(),
-        result.get("creator").asText(),
-        Instant.parse(result.get("created").get(MONGO_DATE_FIELD).asText()),
-        result.get("generator_body"),
-        Instant.parse(result.get("generated").get(MONGO_DATE_FIELD).asText()),
-        result.get("deleted") == null ? Instant.parse(result.get("deleted").get(MONGO_DATE_FIELD).asText()) : null
+        annotation.get("type").asText(),
+        annotation.get("motivation").asText(),
+        annotation.get("target"),
+        annotation.get("body"),
+        annotation.get("preferenceScore").asInt(),
+        annotation.get("creator").asText(),
+        Instant.ofEpochSecond(annotation.get("created").asInt()),
+        annotation.get("generator"),
+        Instant.ofEpochSecond(annotation.get("generated").asInt()),
+        result.get("deleted") == null ? null : Instant.ofEpochSecond(annotation.get("deleted").asInt())
```

Comment:
I like this approach far more than just result

Reply:
Yes so the main thing is that the metadata is generic for each event object. So we have event (metadata) -> eventRecord (metadata of object (pid, version,created) -> Actual object. Based on the SubjectType (an event metadata field) you can know which object is in the eventRecord. So I think it is more generic and more machine readable. Although fieldnames will probably change when we align with Prov-O